test/Microsoft.Azure.Devices.Edge.Test.Common/Linux/EdgeDaemon.cs

```diff
@@ -0,0 +1,209 @@
+                },
+                async () =>
+                {
+                    string[] platformInfo = await Process.RunAsync("lsb_release", "-sir", token);
+                    string os = platformInfo[0].Trim();
+                    string version = platformInfo[1].Trim();
+                    switch (os)
+                    {
+                        case "Ubuntu":
+                            return new[]
+                            {
+                                "set -e",
+                                $"curl https://packages.microsoft.com/config/ubuntu/{version}/prod.list > /etc/apt/sources.list.d/microsoft-prod.list",
+                                "curl https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor > /etc/apt/trusted.gpg.d/microsoft.gpg",
+                                "apt-get update",
+                                "apt-get install --yes iotedge"
+                            };
+                        case "Raspbian":
+                            return new[]
+                            {
+                                "set -e",
+                                "curl -L https://aka.ms/libiothsm-std-linux-armhf-latest -o libiothsm-std.deb",
+                                "curl -L https://aka.ms/iotedged-linux-armhf-latest -o iotedge.deb",
+                                "dpkg --force-confnew -i libiothsm-std.deb iotedge.deb",
+                                "apt-get install -f",
+                                "rm libiothsm-std.deb iotedge.deb"
+                            };
+                        default:
+                            throw new NotImplementedException($"Don't know how to install daemon on operating system '{os}'");
+                    }
+                });
+
+            await Profiler.Run(
+                async () =>
+                {
+                    string[] output = await Process.RunAsync("bash", $"-c \"{string.Join("; ", commands)}\"", token);
+                    Log.Verbose(string.Join("\n", output));
+                },
+                message,
+                properties);
+
+            string hostname = (await File.ReadAllTextAsync("/proc/sys/kernel/hostname", token)).Trim();
+            IotHubConnectionStringBuilder builder = IotHubConnectionStringBuilder.Create(deviceConnectionString);
+
+            await Profiler.Run(
+                async () =>
+                {
+                    await this.InternalStopAsync(token);
+
+                    const string YamlPath = "/etc/iotedge/config.yaml";
+                    string text = await File.ReadAllTextAsync(YamlPath, token);
+
+                    var doc = new YamlDocument(text);
+                    doc.ReplaceOrAdd("provisioning.device_connection_string", deviceConnectionString);
+                    doc.ReplaceOrAdd("hostname", hostname);
+
+                    string result = doc.ToString();
+
+                    FileAttributes attr = 0;
+                    if (File.Exists(YamlPath))
```

Comment:
this if statement should always return true; if file doesn't exit, it will fail when calling File.ReadAllTextAsync at above. right?  should we move this checking up before calling File.ReadAllTextAsync?

Reply:
You're right; I'll remove it.

Reply:
You're right; I'll fix it.

Reply:
Done